src/main/java/net/glowstone/entity/passive/GlowSheep.java

```diff
@@ -123,8 +127,15 @@ public boolean entityInteract(GlowPlayer player, InteractEntityMessage message)
                     Dye dye = (Dye) hand.getData();
                     DyeColor color = dye.getColor();
 
-                    SheepDyeWoolEvent dyeEvent = new SheepDyeWoolEvent(this, color);
+                    SheepDyeWoolEvent dyeEvent = EventFactory.getInstance().callEvent(
+                            new SheepDyeWoolEvent(this, color));
                     if (dyeEvent.isCancelled()) {
+                        List<MetadataMap.Entry> entryList = new ArrayList<>();
+                        entryList.add(new MetadataMap.Entry(
+                                MetadataIndex.SHEEP_DATA, getColorByte()));
+                        player.getSession().send(new EntityMetadataMessage(
```

Comment:
This works, but I would rather have the metadata updated forcefully through the MetadataMap. The metadata is sent to the client by `createUpdateMessage`, which picks up changes inside the map.

What I would do is overload the [`MetadataMap#set`](https://github.com/GlowstoneMC/Glowstone/blob/dev/src/main/java/net/glowstone/entity/meta/MetadataMap.java#L44) method to add a `force` param, which would add the needed field inside the change list even if the [value hasn't changed](https://github.com/GlowstoneMC/Glowstone/blob/dev/src/main/java/net/glowstone/entity/meta/MetadataMap.java#L75-L77).

Let me know if you don't want to do this inside your PR, I will do these changes after your PR is merged otherwise.

Reply:
I see what you mean and I do see the advantages of your suggestion over the former. I have gone ahead and made the change.